targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/common_rtc.h

```diff
@@ -45,6 +45,7 @@ extern bool              m_common_rtc_enabled;
 extern uint32_t volatile m_common_rtc_overflows;
 extern bool volatile     lp_ticker_interrupt_fire;
 
+void     common_rtc_free();
```

Comment:
I'd expect this to be still warning you. You need `(void)` in C.

Reply:
No warning after the change. Is it perhaps because a C++ compiler is used?

Reply:
It is legal in C, so a compiler may not warn, even in C mode.

But it's an old-style (pre-C90) "unspecified parameters" declaration. You don't mean that - you do specifically want to declare it as taking no parameters.

Calling `common_rtc_free(3)` is valid with that declaration. Wouldn't be with `(void)`.

Reply:
Just double-checked that they didn't make any changes in C11. Apparently not - still the same as C99:

> The use of function declarators with empty parentheses (not prototype-format parameter
type declarators) is an obsolescent feature.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/849a22cfd64f9164f27e4da4f8e1a88a74115d6f..992133be50e2567d8742bfb5a9cc32ec6bcf6df8).